pandas/core/frame.py

```diff
@@ -6240,8 +6240,10 @@ def explode(self, column: Union[str, Tuple]) -> "DataFrame":
         result = self[column].explode()
         return (
             self.drop([column], axis=1)
-            .join(result)
+            .reset_index(drop=True)
+            .join(self[column].reset_index(drop=True).explode())
```

Comment:
I don't think we need to `explode` the `Series` twice if you don't drop the index.
```python
result = self[column].reset_index().explode()
return (
    self.drop([column], axis=1)
    .reset_index(drop=True)
    .join(result[column])
    .reindex(columns=self.columns, copy=False)
    .set_index(result[index_col])
)
```

I guess it's not immediate to get `index_col` since the index can have a name or not, but I think it's worth avoiding the double `explode`.

Reply:
Thanks for your review!

That's a good point, thanks - I'll work on this